File: Programming EF Code First/chapter5/Chapter 5 VB/Data Annotations/Model/Lodging.vb
Imports System.ComponentModel.DataAnnotations

Namespace Model
  Public Class Lodging
	Public Property LodgingId() As Integer
	<Required, MaxLength(200), MinLength(10)>
	Public Property Name() As String
	Public Property Owner() As String
	Public Property MilesFromNearestAirport() As Decimal

	<Column("destination_id")>
	Public Property DestinationId() As Integer
	Public Property Destination() As Destination
	Public Property InternetSpecials() As List(Of InternetSpecial)
	Public Property PrimaryContactId() As Integer?
	<InverseProperty("PrimaryContactFor"), ForeignKey("PrimaryContactId")>
	Public Property PrimaryContact() As Person
	Public Property SecondaryContactId() As Integer?
	<InverseProperty("SecondaryContactFor"), ForeignKey("SecondaryContactId")>
	Public Property SecondaryContact() As Person
  End Class
End Namespace

File: Programming EF Code First/chapter5/Chapter5 CSharp/BAGA_DataAnnotations/Model/Lodging.cs
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;
using System;

namespace Model
{
  public class Lodging
  {
    public int LodgingId { get; set; }
    [Required]
    [MaxLength(200)]
    [MinLength(10)]
    public string Name { get; set; }
    public string Owner { get; set; }
    public decimal MilesFromNearestAirport { get; set; }

    [Column("destination_id")]
    public int DestinationId { get; set; }
    public Destination Destination { get; set; }
    public List<InternetSpecial> InternetSpecials { get; set; }
    public Nullable<int> PrimaryContactId { get; set; }
    [InverseProperty("PrimaryContactFor")]
    [ForeignKey("PrimaryContactId")]
    public Person PrimaryContact { get; set; }
    public Nullable<int> SecondaryContactId { get; set; }
    [InverseProperty("SecondaryContactFor")]
    [ForeignKey("SecondaryContactId")]
    public Person SecondaryContact { get; set; }
  }
}